### PR TITLE
Add KafkaSource dispatcher StatefulSet manifest

### PR DESCRIPTION
**Pull request by pierDipi**
As per title.

- Copy `kafka-source-dispatcher` deployment
- Make it a StatefulSet

### CODE REVIEW

File: data-plane/config/sourcev2/500-dispatcher.yaml [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+          terminationMessagePath: /dev/temination-log
+          securityContext:
+            allowPrivilegeEscalation: false
+            privileged: false
+            readOnlyRootFilesystem: true
+      volumes:
+        - name: config-kafka-source-data-plane
+          configMap:
+            name: config-kafka-source-data-plane
+        - name: cache
```
Comment (aavarghese) [inline data-plane/config/sourcev2/500-dispatcher.yaml:146]: Don't we need a volume here for `contract-resources`?

Comment (pierDipi) [reply data-plane/config/sourcev2/500-dispatcher.yaml:146]: That will be injected/created by the Pod webhook, see https://github.com/knative-sandbox/eventing-kafka-broker/pull/1796

Comment (aavarghese) [reply data-plane/config/sourcev2/500-dispatcher.yaml:146]: Perfect. Thanks for making the connection! Will TAL at that one

**Review by aavarghese — APPROVED** [review]
/lgtm

**Comment by matzew** [comment]
`deployment` being "deprecated" ?

**Comment by pierDipi** [comment]
> `deployment` being "deprecated" ?

Yes, next PRs

**Pull request merged**